### PR TITLE
added ores and ORM

### LINKED ISSUE #2301: Minerals B: 70
# Description
Change the logic for mineable rocks so that the minerals they contained is determined randomly on map load, not when the rock is mined.

Add all the different kinds of minerals and ores that can be mined with their appropriate rarity. This should use a prefab variant most likely. The ores should also be stackable (add Stackable component).

You can skip gibtonite for this ticket.

In TG, logic for minerals can be found in minerals.dm, for example spawn chance config is in mineralSpawnChanceList . It looks like there are multiple spawn chance lists defined so they can either be randomly swapped between or configured for a server. 

Accordingly, the spawn chance list for unitystation should be defined using an SO so it can be swapped out with others. It can be referenced from a singleton SO or manager to define which one to use when the map loads.

### PR DESCRIPTION
**Pull request by Bod9001**
### Purpose
Fixes #2301
addes Minerals that randomly generate on asteroids On round start and I did a working ORM to cargo ( no UI )
### Open Questions and Pre-Merge TODOs

- [x]  This fix is tested on the same branch it is PR'ed to.
- [?]  I correctly commented my code
- [x]  My code is indented with tabs and not spaces
- [x]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [x]  Any new files are named using PascalCase (to avoid issues on case sensitive file systems)
- [?]  Any new / changed components follow the [Component Development Checklist](https://github.com/unitystation/unitystation/wiki/Component-Development-Checklist)
- [x]  Any new objects / items follow the [Creating Items and Objects Guide](https://github.com/unitystation/unitystation/wiki/Creating-Items-and-Objects%3A-Now-With-Prefab-Variants) (especially concerning the use of prefab variants)
- [x]  This PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)
- [ ]  This PR has been tested with round restarts.
- [ ]  This PR has been tested on moving / rotating / rotated-before-joining matrices (if applicable)

### CODE REVIEW

File: UnityProject/Assets/Scripts/Mining/OreRedemptionMachine.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+			Logger.Log("2" + Ore.ToString());
+			foreach (var exOre in ExpectedOres) {
+				Logger.Log("3");
+				if (Ore != null)
+				{
+					if (Ore.HasTrait(exOre.Tray))
+					{
+						Logger.Log("4");
+						var inStackable = Ore.gameObject.GetComponent<Stackable>();
+						Spawn.ServerPrefab(exOre.Material, RegisterObject.WorldPositionServer + Vector3Int.down, transform.parent, count: inStackable.Amount );
```
Comment (chairbender) [inline UnityProject/Assets/Scripts/Mining/OreRedemptionMachine.cs]: I'm guessing you already know this but just to note I think the sheet recipes are a bit more complicated than this, like you have some sheets which take multiple ores, but this is fine for this PR, we can tackle that in the ORM UI

File: UnityProject/Assets/Scripts/Ore Generation/OreGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+				NodeScatter(OreTile.Location, OreCategorie);
+			}
+		}
+	}
+
+	void NodeScatter(Vector3Int Location, WeightNStrength MaterialSpecified)
+	{
+
+
+		var Locations = new List<Vector3Int>() {
```
Comment (chairbender) [inline UnityProject/Assets/Scripts/Ore Generation/OreGenerator.cs]: What's the purpose of this and ChosenLocation? ChosenLocation is written to but never read.

**Review by chairbender — CHANGES_REQUESTED** [review]
Awesome, was quite excited to see this pop up and it's working great. The way ore generation probabilities work is pretty creative, it was a little confusing to wrap my head around at first but it makes sense now and is leading to some nice looking patterns.

I have quite a few renamings, refactorings, public vs. private changes to request and I know those are quite annoying for you to fix, so instead I've submitted a PR to your branch in your own fork that fixes them: https://github.com/Bod9001/unitystation/pull/2

 So, next steps:

1. Please merge my PR and make sure it's still working how you expect.
2. Please also merge latest develop into your branch so it can be tested with the latest networking fixes. I want to test and make sure it works fine on server / headless / round restarts with the latest networking fixes in place. 
3. Take a look at below comments

File: UnityProject/Assets/Scripts/Mining/OreRedemptionMachine.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Mining/OreRedemptionMachine.cs]: yeah, Meant to be temporary

File: UnityProject/Assets/Scripts/Ore Generation/OreGenerator.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Ore Generation/OreGenerator.cs]: Let me Swap out Location  and ChosenLocation  It's supposed to be using that

**Pull request merged**